### ISSUE TITLE
Trying to get property of non-object

### ISSUE THREAD

**Issue by LordA98** [versionpress/support#164]
Environment Details:
- VersionPress version: 4.0-beta
- WordPress version: 4.9.9
- OS: Windows NT ALEX-PC 10.0 build 17134 (Windows 10) AMD64
- Webserver: Apache & MySQL Webservers on WAMP
- PHP Version: 7.1.23
- Git version: 2.19.0

I'm creating a custom plugin for a site I'm working on.

When activating my plugin, I get the `The plugin generated 243 characters of unexpected output during activation` error.

In the debug.log file, it says `[25-Feb-2019 21:13:08 UTC] PHP Notice:  Trying to get property of non-object in C:\Bitnami\wampstack-7.1.23-0\apache2\htdocs\wordpress\wp-content\plugins\versionpress\vendor\phpmyadmin\sql-parser\src\Components\OptionsArray.php on line 218`.

It gives these errors when running the WordPress dbdelta() function with this SQL passed in:

CREATE TABLE tablename (
    id mediumint(9) NOT NULL AUTO_INCREMENT,
    name VARCHAR(20) NOT NULL,
    rank int(3) NOT NULL,
    date_created datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
    date_updated datetime, 
    author int(3) NOT NULL,
    editor int(3),
    side boolean NOT NULL DEFAULT 0,
    toPublish boolean NOT NULL DEFAULT 1,
    parent_menu mediumint(9) NOT NULL,
    PRIMARY KEY  (id),
    FOREIGN KEY (parent_menu) REFERENCES foreigntable(id)
) utf8mb4_unicode_520_ci;

Strangely enough, deactivating VersionPress still gives the same error when activating my plugin.

The dbdelta function is used 3 other times at the same point to create 3 other tables, but only fails on this one. 

Happy to give more information if needed.

Thanks in advance.

**Comment by pavelevap** [comment]
Hi @LordA98,

maybe your query is missing `COLLATE` statement in the end?

```
CREATE TABLE tablename (
  id mediumint(9) NOT NULL AUTO_INCREMENT,
  name VARCHAR(20) NOT NULL,
  rank int(3) NOT NULL,
  date_created datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
  date_updated datetime,
  author int(3) NOT NULL,
  editor int(3),
  side boolean NOT NULL DEFAULT 0,
  toPublish boolean NOT NULL DEFAULT 1,
  parent_menu mediumint(9) NOT NULL,
  PRIMARY KEY (id),
  FOREIGN KEY (parent_menu) REFERENCES foreigntable(id)
) COLLATE utf8mb4_unicode_520_ci;
```

If you can share your script (or part of it), I can test it and see if it is somehow related to VersionPress.

**Comment by LordA98** [comment]
Hi @pavelevap,

Thanks for the reply.  Sorry for the delay.

Adding COLLATE didn't solve the error.

Upon activation, this function gets called:
```
function activate() {
        // Create DB Tables
        $this->db->createTables();

        flush_rewrite_rules();
	}
```

createTables calls this function 4 times (one for each table needed for the plugin):
```private function createTable($table) {
        // Get correct SQL file
        $sql = file_get_contents( WP_PLUGIN_DIR  . "/lion-menu/assets/sql/create_" . $table . "_table.sql" );

        // Set table name
        $sql = str_replace("tableplaceholder", $this->wpdb->prefix . "lm_" . $table, $sql);

        // If sql contains a foreign key - add prefix
        if (strpos($sql, 'FOREIGN KEY') !== false) {
            $sql = str_replace("prefixplaceholder", $this->wpdb->prefix . "lm", $sql);
        }

        // Set charset 
        $sql = str_replace("charsetplaceholder", $this->wpdb->get_charset_collate(), $sql);

        // log_me($sql);

        dbDelta($sql);
    }
```

The dbDelta() function at the end is causing the issue. 

Let me know if you need more information / code.

I haven't been looking into this as I've been busy, but I will also try to fix it on my end as well.

Thank you,
Alex

**Comment by LordA98** [comment]
Hi, 

I had given up and accepted that this issue was going to persist.  

I've just accidentally stumbled across the solution.

It appears that using DEFAULT 0 causes there to be unexpected output and gives the 'Trying to get property of non-object' error in the debug.log.

Removing this fixes the issue.

I have no idea why DEFAULT 0 is causing the error but at least there's somewhat of a fix.

Thanks for taking a look at the issue anyway.

**Comment by pavelevap** [comment]
Hi, @LordA98,

thank you for your answer. Could you please try our latest [4.0-beta2](https://github.com/versionpress/versionpress/releases/tag/4.0-beta2) version and test it also with your `DEFAULT 0` example? If I identified cause of this problem correctly, it should work now.

**Comment by pavelevap** [comment]
I did not try it, but my best guess is issue related to SQL Parser: https://github.com/phpmyadmin/phpmyadmin/issues/12165

It was fixed in version 3.4 and `4.0-beta2` should use this version: https://github.com/versionpress/versionpress/commit/f28b428b49a43f4c6268d1f1b19c0c3f0330d7e8#diff-26a62a4574c0de4eaaacd51b04dd27e8.
